lib/shared/stats.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Provides helper functions to collect, store, and receive time measurements
+ * and pass the final statistics object to the lint result for the 'stats' ESLint option.
+ * @author Mara Kiefer <http://github.com/mnkiefer>
+ */
+"use strict";
+
+module.exports = {
+
+    /**
+     * Start time measurement
+     * @returns {number} t variable for tracking time
+     */
+    startTime() {
+        return process.hrtime();
+    },
+
+    /**
+     * End time measurement
+     * @param {number} t Variable for tracking time
+     * @returns {number} The measured time in milliseconds
+     */
+    endTime(t) {
+        const time = process.hrtime(t);
+
+        return time[0] * 1e3 + time[1] / 1e6;
+    },
+
+    /**
+     * Get statistics for lint result (see 'stats' ESLint option)
+     * @param {Object} linter Linter instance
+     * @param {string} filename File that is currently being linted
+     * @returns {Object} Statistics result object
+     */
+    getStats(linter, filename) {
```

Comment:
It might be worth passing in just the fix passes and times as separate arguments rather than passing in `linter`. This would make it easier to test this method.

Reply:
You are right! 👍 Looking at it now after the refactoring, I also realized that we no longer need the `getStats()` function and just call:
```js
result.stats = {
    times: linter.getTimes(filePath),
    fixPasses: linter.getFixPasses()
};
```
See https://github.com/eslint/eslint/pull/17850/commits/762c74dcd2e97070ac76dbe5120115fb33094c8f, where I've also updated the header comment to account for this.